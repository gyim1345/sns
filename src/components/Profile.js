import React from 'react';

const Profile = ({ id, name }) => {
  return (
    <div>
      <b>{id}</b>&nbsp;
      <span>{name}</span>
    </div>
  );
};

export default Profile;
