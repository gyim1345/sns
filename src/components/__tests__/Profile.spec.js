import React from 'react';
import { mount } from 'enzyme';
import Profile from '../Profile';

describe('<Profile />', () => {
  it('스냅샷 비교', () => {
    const wrapper = mount(<Profile id={'id'} name={'kim'} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('렌더 props 비교', () => {
    const wrapper = mount(<Profile id={'id'} name={'kim'} />);
    console.log(wrapper.props());
    expect(wrapper.props().id).toBe('id');
    expect(wrapper.props().name).toBe('kim');

    const boldElement = wrapper.find('b'); // find(): 특정 DOM 선택 가능 querySelector같음
    expect(boldElement.contains('id')).toBe(true); //contains(): innerText같은 안의 값이 id인지
    const spanElement = wrapper.find('span');
    expect(spanElement.text()).toBe('kim'); //text(): span의 innerText 같은 기능.
  });
});
