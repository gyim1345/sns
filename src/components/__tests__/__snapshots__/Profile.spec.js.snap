// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> 스냅샷 비교 1`] = `
<Profile
  id="id"
  name="kim"
>
  <div>
    <b>
      id
    </b>
     
    <span>
      kim
    </span>
  </div>
</Profile>
`;
